specification/operationalinsights/data-plane/Microsoft.OperationalInsights/preview/2021-05-19_Preview/OperationalInsights.json

```diff
@@ -123,6 +123,78 @@
         }
       }
     },
+    "/{resourceUri}/query": {
+      "get": {
+        "operationId": "Query_ResourceGet",
+        "summary": "Execute an Analytics query using resource URI",
+        "description": "Executes an Analytics query for data in the context of a resource. [Here](https://docs.microsoft.com/azure/azure-monitor/logs/api/azure-resource-queries) is an example for using POST with an Analytics query.",
+        "x-ms-examples": {
+          "simple-query": {
+            "$ref": "examples/oms-get-resource-centric-example.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "#/parameters/queryParam"
+          },
+          {
+            "$ref": "#/parameters/timespanParam"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The API call succeeded and the Analytics query result is in the response payload",
+            "schema": {
+              "$ref": "#/definitions/queryResults"
+            }
+          },
+          "default": {
+            "description": "An error response object.",
+            "schema": {
+              "$ref": "#/definitions/errorResponse"
+            }
+          }
+        }
+      },
+      "post": {
+        "operationId": "Query_ResourcePost",
```

Comment:
To be consistent with the workspace query operation id, we should call this `Query_ResourceExecute`.